### PR TITLE
Adding onChange events in editable dropdowns for values changed through code.

### PR DESCRIPTION
**Pull request by aasimkhan30**
I have added a small example in migration extension

This will make it consistent will non editable version of dropdown where an event is generated when we update the value through code.

### CODE REVIEW

File: src/sql/workbench/browser/modelComponents/dropdown.component.ts [hunk 1]
```diff
@@ -141,7 +141,13 @@ export default class DropDownComponent extends ComponentBase<azdata.DropDownProp
 		if (this.editable && !this._isInAccessibilityMode) {
 			this._editableDropdown.values = this.getValues();
 			if (this.value) {
-				this._editableDropdown.value = this.getSelectedValue();
+				if (this._editableDropdown.value !== this.getSelectedValue()) {
```
Comment (alanrenmsft) [inline src/sql/workbench/browser/modelComponents/dropdown.component.ts:144]: the event should be fired by the editableDropdown itself, right? here is just a angular wrapper of it.

**Review by alanrenmsft — APPROVED** [review]
looks good to me, please test the places where we are using this component and make sure everything is fine.

File: src/sql/base/parts/editableDropdown/browser/dropdown.ts [hunk 2]
```diff
@@ -263,9 +263,11 @@ export class Dropdown extends Disposable implements IListVirtualDelegate<string>
 		this._selectListContainer.setAttribute('aria-hidden', `${!visible}`);
 	}
 
-	private _updateSelection(newValue: string): void {
-		this.value = newValue;
-		this._onValueChange.fire(newValue);
+	public updateSelection(newValue: string): void {
+		if (this.value !== newValue) {
+			this.value = newValue;
+			this._onValueChange.fire(newValue);
```
Comment (Charles-Gagnon) [inline src/sql/base/parts/editableDropdown/browser/dropdown.ts:269]: I would think that setting value should fire the event. Why would we only fire this when updateSelection is called specifically?

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]

**Review by alanrenmsft — APPROVED** [review]

**Comment by alanrenmsft** [comment]
tests are failing, related to your change.

**Comment by Charles-Gagnon** [comment]
It looks like there's an error that's happening when dropdown values are selected too : 

`Cannot read property 'map' of undefined: TypeError: Cannot read property 'map' of undefined`

Although i do seem to be able to repro this on stable as well...but while you're working on this could you see what's going on there too?

**Comment by Charles-Gagnon** [comment]
@aasimkhan30 FYI - in case you end up looking at that map issue I created a separate issue for it for better tracking : https://github.com/microsoft/azuredatastudio/issues/14916

**Pull request closed without merging**